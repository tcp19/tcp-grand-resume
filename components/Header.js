import {
    Box,
    Flex,
    Text,
    IconButton,
    Button,
    Stack,
    Collapse,
    Icon,
    Link,
    useColorModeValue,
    useBreakpointValue,
    useDisclosure,
} from "@chakra-ui/react";
import {
    HamburgerIcon,
    CloseIcon,
    ChevronDownIcon,
    ChevronRightIcon,
} from "@chakra-ui/icons";
import RouteLink from "next/link";
import { useEffect, useState } from "react";
import { useRouter } from "next/router";

export default function Header() {
    const { isOpen, onToggle } = useDisclosure();
    const [showGetStartedButton, setShowGetStartedButton] = useState(true);

    const router = useRouter();

    useEffect(() => {
        if (router.route !== "/") {
            setShowGetStartedButton(false)
        } else {
            setShowGetStartedButton(true)
        }
    }, [router.route]);

    return (
        <Box>
            <Flex
                bg={useColorModeValue("white", "gray.800")}
                color={useColorModeValue("gray.600", "white")}
                minH={"60px"}
                py={{ base: 2 }}
                px={{ base: 4 }}
                borderBottom={1}
                borderStyle={"solid"}
                borderColor={useColorModeValue("gray.200", "gray.900")}
                align={"center"}
            >
                <Flex
                    flex={{ base: 1, md: "auto" }}
                    ml={{ base: -2 }}
                    display={{ base: "flex", md: "none" }}
                >
                    <IconButton
                        onClick={onToggle}
                        icon={
                            isOpen ? (
                                <CloseIcon w={3} h={3} />
                            ) : (
                                <HamburgerIcon w={5} h={5} />
                            )
                        }
                        variant={"ghost"}
                        aria-label={"Toggle Navigation"}
                    />
                </Flex>
                <Flex
                    w={"250px"}
                    flex={{ base: 1 }}
                    justify={{ base: "center", md: "start" }}
                >
                    <Text
                        fontSize="2xl"
                        textAlign={useBreakpointValue({
                            base: "center",
                            md: "left",
                        })}
                        fontFamily={"heading"}
                        className="u-stroke"
                        color={useColorModeValue("gray.800", "white")}
                    >
                        Grand resume
                    </Text>
                </Flex>
                {
                    showGetStartedButton && (
                        <Stack
                            flex={{ base: 1, md: 0 }}
                            justify={"flex-end"}
                            direction={"row"}
                            spacing={6}
                        >
                            <RouteLink href="/build-resume">
                                <Button
                                    display={{ base: "none", md: "inline-flex" }}
                                    fontSize={"sm"}
                                    fontWeight={600}
                                    color={"white"}
                                    bg={"#181C27"}
                                    href={"#"}
                                    _hover={{
                                        bg: "#181C27",
                                    }}
                                >
                                    Get started
                        </Button>
                            </RouteLink>
                        </Stack>
                    )
                }

            </Flex>

            <Collapse in={isOpen} animateOpacity>
                <MobileNav />
            </Collapse>
        </Box>
    );
}

const MobileNav = () => {
    return (
        <Stack
            bg={useColorModeValue("white", "gray.800")}
            p={4}
            display={{ md: "none" }}
        >
            {NAV_ITEMS.map((navItem) => (
                <MobileNavItem key={navItem.label} {...navItem} />
            ))}
        </Stack>
    );
};

const MobileNavItem = ({ label, children, href }) => {
    const { isOpen, onToggle } = useDisclosure();

    return (
        <Stack spacing={4} onClick={children && onToggle}>
            <Flex
                py={2}
                as={Link}
                href={href ?? "#"}
                justify={"space-between"}
                align={"center"}
                _hover={{
                    textDecoration: "none",
                }}
            >
                <Text
                    fontWeight={600}
                    color={useColorModeValue("gray.600", "gray.200")}
                >
                    {label}
                </Text>
                {children && (
                    <Icon
                        as={ChevronDownIcon}
                        transition={"all .25s ease-in-out"}
                        transform={isOpen ? "rotate(180deg)" : ""}
                        w={6}
                        h={6}
                    />
                )}
            </Flex>

            <Collapse
                in={isOpen}
                animateOpacity
                style={{ marginTop: "0!important" }}
            >
                <Stack
                    mt={2}
                    pl={4}
                    borderLeft={1}
                    borderStyle={"solid"}
                    borderColor={useColorModeValue("gray.200", "gray.700")}
                    align={"start"}
                >
                    {children &&
                        children.map((child) => (
                            <Link key={child.label} py={2} href={child.href}>
                                {child.label}
                            </Link>
                        ))}
                </Stack>
            </Collapse>
        </Stack>
    );
};

const NAV_ITEMS = [
    {
        label: "Templates",
        children: [
            {
                label: "Danfo",
                subLabel: "Trending Design to inspire you",
                href: "#",
            },
            {
                label: "Google ",
                subLabel: "Up-and-coming Designers",
                href: "#",
            },
            {
                label: "Uber",
                // subLabel: 'Up-and-coming Designers',
                href: "#",
            },
        ],
    },
    {
        label: "Contribute a Template",
        href: "#",
    },
];
