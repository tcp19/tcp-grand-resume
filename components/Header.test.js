import "../matchMedia.mock";
import { render } from "@testing-library/react";
import Header, { MobileNav, MobileNavItem } from "./Header";

describe("Header", () => {
    it("should render the component properly", () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });

    it("should render the MobileNav component properly", () => {
        const { container } = render(<MobileNav />);
        expect(container).toMatchSnapshot();
    });

    it("should render the MobileNavItem component properly", () => {
        const { container } = render(
            <MobileNavItem label="Contribute a Template" href="#" key={1} />
        );
        expect(container).toMatchSnapshot();
    });
});
