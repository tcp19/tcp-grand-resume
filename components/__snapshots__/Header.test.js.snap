// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the MobileNav component properly 1`] = `
<div>
  <div
    class="chakra-stack css-tm2a1"
  >
    <div
      class="chakra-stack css-17kdbol"
    >
      <a
        class="chakra-link css-q4j97j"
        href="#"
      >
        <p
          class="chakra-text css-y33gq0"
        >
          Templates
        </p>
        <svg
          class="chakra-icon chakra-icon css-1bfsybq"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </a>
      <div
        class="chakra-collapse"
        style="overflow: hidden; display: none; opacity: 0; height: 0px;"
      >
        <div
          class="chakra-stack css-khmogt"
        >
          <a
            class="chakra-link css-o5g2qp"
            href="#"
          >
            Danfo
          </a>
          <a
            class="chakra-link css-o5g2qp"
            href="#"
          >
            Google 
          </a>
          <a
            class="chakra-link css-o5g2qp"
            href="#"
          >
            Uber
          </a>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-17kdbol"
    >
      <a
        class="chakra-link css-q4j97j"
        href="#"
      >
        <p
          class="chakra-text css-y33gq0"
        >
          Contribute a Template
        </p>
      </a>
      <div
        class="chakra-collapse"
        style="overflow: hidden; display: none; opacity: 0; height: 0px;"
      >
        <div
          class="chakra-stack css-khmogt"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Header should render the MobileNavItem component properly 1`] = `
<div>
  <div
    class="chakra-stack css-17kdbol"
  >
    <a
      class="chakra-link css-q4j97j"
      href="#"
    >
      <p
        class="chakra-text css-y33gq0"
      >
        Contribute a Template
      </p>
    </a>
    <div
      class="chakra-collapse"
      style="overflow: hidden; display: none; opacity: 0; height: 0px;"
    >
      <div
        class="chakra-stack css-khmogt"
      />
    </div>
  </div>
</div>
`;

exports[`Header should render the component properly 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-8lthux"
    >
      <div
        class="css-1fsntt9"
      >
        <button
          aria-label="Toggle Navigation"
          class="chakra-button css-198bfjk"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-w31a5m"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="css-1r8knn8"
      >
        <p
          class="chakra-text u-stroke css-1i519ap"
        >
          Grand resume
        </p>
      </div>
      <div
        class="chakra-stack css-18njwrs"
      >
        <button
          class="chakra-button css-149nlxd"
          href="#"
          type="button"
        >
          Get started
        </button>
      </div>
    </div>
    <div
      class="chakra-collapse"
      style="overflow: hidden; display: none; opacity: 0; height: 0px;"
    >
      <div
        class="chakra-stack css-tm2a1"
      >
        <div
          class="chakra-stack css-17kdbol"
        >
          <a
            class="chakra-link css-q4j97j"
            href="#"
          >
            <p
              class="chakra-text css-y33gq0"
            >
              Templates
            </p>
            <svg
              class="chakra-icon chakra-icon css-1bfsybq"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </a>
          <div
            class="chakra-collapse"
            style="overflow: hidden; display: none; opacity: 0; height: 0px;"
          >
            <div
              class="chakra-stack css-khmogt"
            >
              <a
                class="chakra-link css-o5g2qp"
                href="#"
              >
                Danfo
              </a>
              <a
                class="chakra-link css-o5g2qp"
                href="#"
              >
                Google 
              </a>
              <a
                class="chakra-link css-o5g2qp"
                href="#"
              >
                Uber
              </a>
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-17kdbol"
        >
          <a
            class="chakra-link css-q4j97j"
            href="#"
          >
            <p
              class="chakra-text css-y33gq0"
            >
              Contribute a Template
            </p>
          </a>
          <div
            class="chakra-collapse"
            style="overflow: hidden; display: none; opacity: 0; height: 0px;"
          >
            <div
              class="chakra-stack css-khmogt"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
